src/app/static/src/app/components/files/FileUpload.vue

```diff
@@ -0,0 +1,92 @@
+<template>
+    <div>
+        <b-dropdown text="Select file from" variant="white">
+            <a class="dropdown-item" href="#" v-on:mousedown="$refs[name].click()">
+                <input type="file"
+                       style="display:none"
+                       :ref="name"
+                       :id="name"
+                       :accept="accept"
+                       @click="$emit('input-opened')"
```

Comment:
Er, no, not even sure what I was trying there.

Reply:
Ah sorry, I realised what this is for. It's in there purely for testing purposes, to test that a click on this hidden file input is triggered when the dropdown link is clicked.  It's really not ideal but I can't figure out another way to verify this? Checking whether the input has focus doesn't work because it's a hidden input.